### PR TITLE
Write Warehouse Delegate, revise BPMNs

### PR DESCRIPTION
**Pull request by VitalyVakhteev**
Query:

293438284-a3f67a82-add4-4815-972c-7694c2a8c171

Dashboard:

293438287-587687df-8238-4be1-aeca-04122c78f6de

### CODE REVIEW

File: engine/src/main/java/ai/hhrdr/chainflow/engine/WarehouseDelegate.java [hunk 1]
```diff
@@ -2,18 +2,32 @@
...
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
 import java.util.logging.Logger;
 
 /**
  * This is an easy adapter implementation
  * illustrating how a Java Delegate can be used
  * from within a BPMN 2.0 Service Task.
  */
+@Component("warehouse")
```
Comment (evahteev) [inline engine/src/main/java/ai/hhrdr/chainflow/engine/WarehouseDelegate.java:16]: \\https://www.baeldung.com/spring-component-annotation

File: engine/src/main/java/ai/hhrdr/chainflow/engine/WarehouseDelegate.java [hunk 2]
```diff
@@ -26,10 +40,36 @@ public void execute(DelegateExecution execution) throws Exception {
             + ", variables=" + execution.getVariables()
             + " \n\n");
 
-    // Get ID, type of publication (dashboard, query, etc.) make post request to change is_draft to false
-    // Precondition: Complete is successful
-    //    execution.getVariables();
-
+    String json = "{\"is_draft\":false}";
+    HttpClient client = HttpClient.newHttpClient();
+    if (execution.getVariable("dashboard_id") != null) {
```
Comment (evahteev) [inline engine/src/main/java/ai/hhrdr/chainflow/engine/WarehouseDelegate.java]: switch

Comment (evahteev) [reply engine/src/main/java/ai/hhrdr/chainflow/engine/WarehouseDelegate.java]: switch from bpmn name, not from variables.

File: engine/src/main/java/ai/hhrdr/chainflow/engine/WarehouseDelegate.java [hunk 3]
```diff
@@ -26,10 +40,36 @@ public void execute(DelegateExecution execution) throws Exception {
...
+      sendRequest(client, req);
+    } else if (execution.getVariable("query_id") != null) {
+      HttpRequest req = HttpRequest
+              .newBuilder()
+              .uri(URI.create(apiURL + "/api/queries/" + execution.getVariable("query_id")))
+              .header("Content-Type", "application/json")
+              .header("Authorization", apiKey)
+              .POST(HttpRequest.BodyPublishers.ofString(json))
+              .build();
+      sendRequest(client, req);
```
Comment (evahteev) [inline engine/src/main/java/ai/hhrdr/chainflow/engine/WarehouseDelegate.java]: try catch here, as results can be anything

File: engine/src/main/resources/application.properties [hunk 4]
```diff
@@ -2,3 +2,6 @@ ethereum.privateKey=${ETHEREUM_PRIVATEKEY:0x611f00b791a44ccde1158140b9ec5360b2a1
 ethereum.defaultFundingCommitment=${DEFAULT_FUNDING:1000}
 ethereum.rpcUrl=${RPC_URL:https://rpc.ankr.com/polygon_mumbai}
 ethereum.factoryAddress=${FACTORY_ADDRESS:0x8E1c92D50c4A9DD7ef46C3d77Db0A7Cb6D300f86}
+api.url=http://api-url.com
```
Comment (evahteev) [inline engine/src/main/resources/application.properties]: ${API_URL:http://api-url.com}

**Comment by evahteev** [comment]
@VitalyVakhteev add screenshots for bmpns from modeler for review

**Review by evahteev — CHANGES_REQUESTED** [review]

**Comment by VitalyVakhteev** [comment]
Query:

warehouse_query_review

Dashboard:

warehouse_dashboard_review

File: engine/src/main/java/ai/hhrdr/chainflow/engine/WarehouseDelegate.java (hunk 2 above)
Comment (VitalyVakhteev) [reply engine/src/main/java/ai/hhrdr/chainflow/engine/WarehouseDelegate.java]: Should I use getKey or getResourceName? Key has only the name for the bpmn, while resourceName includes the file extension. So far, I've written it to use the key only.

**Review by evahteev — APPROVED** [review]
LGTM

**Pull request merged**